experimental/cache_size.cc

```diff
@@ -0,0 +1,70 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <random>
+
+#include "../demos/asm/measurereadlatency.h"
+#include "../demos/instr.h"
+#include "../demos/utils.h"
+
+// Determines the maximum latency of reading elements of a vector of size "sz",
+// which is given as input.
+// It reads each element of the vector and measures the time
+// using MeasureReadLatency(). Later this number can be used to determine cache
+// misses and hits
+
+uint64_t AnalyzeReadingTime(int64_t sz) {
+  // Generate a random order of accesses to eliminate the impact of hw
+  // prefetchers
+  std::vector<int64_t> accesses(sz);
+  for (int64_t i = 0; i < sz; i++) {
+    accesses.push_back(i);
+  }
+  std::mt19937 f(std::random_device());
```

Comment:
Oh yo I don't think this works. `std::mt19937` takes its parameter by reference, so the code snippet I gave won't work and neither should this -- a temporary (like the result of evaluating `std::random_device()`) can't be passed in as a non-const reference parameter.

This line will actually compile, but incorrectly, due to the Most Vexing Parse. And that shooooould lead the next line to fail horribly.

Anyway, `std::random_device()` should be stored in a local variable.

Reply:
Yup, I fixed it in the main code.

Reply:
Just a sudden change at the end when I was reviewing PR comments! 
I suddenly decided to changed it based on the code snippet you added (that was bad I know :( )